Website/Default.aspx.cs

```diff
@@ -722,7 +725,47 @@ protected override void OnInit(EventArgs e)
 		        AJAX.GetScriptManager(this).AsyncPostBackTimeout = Host.AsyncTimeout;
 	        }
         }
-        
+
+        private string GenerateCssCustomProperties()
+        {
+            string cacheKey = $"Dnn_Css_Custom_Properties_{PortalSettings.PortalId}";
+            string cache = Common.Utilities.DataCache.GetCache<string>(cacheKey);
+            if (string.IsNullOrEmpty(cache))

```

Comment:
Current control flow results in two hits to cache per page load.  I'd recommend switching the control logic here, which also reduces IL nesting which is important for heavy load items.

Just switch to 

if(!string.IsNullorEmpty(cache))
    return cache;

//Otherwise continue with the other stuff